ext/openssl/ossl_engine.c

```diff
@@ -94,8 +104,12 @@ ossl_engine_s_load(int argc, VALUE *argv, VALUE klass)
     StringValueCStr(name);
 #ifndef OPENSSL_NO_STATIC_ENGINE
 #if HAVE_ENGINE_LOAD_DYNAMIC
+#if !defined(LIBRESSL_VERSION_NUMBER) && (OPENSSL_VERSION_NUMBER >= 0x10100000L)
+    OSSL_ENGINE_LOAD_IF_MATCH(DYNAMIC);
+#else
     OSSL_ENGINE_LOAD_IF_MATCH(dynamic);
 #endif
+#endif
```

Comment:
This is comparing the string with "DYNAMIC" rather than "dynamic". The return value from OPENSSL_init_crypto() should be checked.